components/ModEditSubmissionForm.vue

```diff
@@ -723,54 +739,43 @@ watch(moderationSubmissionStore, newValue => {
 
 watch(moderationSubmissionStore, newValue => {
     moderationSubmissionStore.filterSelectedSubmission(newValue.selectedSubmissionId)
-    autofillEditSubmissionForm(newValue.selectedSubmissionData)
+    initializeSubmissionFormValues(newValue.selectedSubmissionData)
 })
 
 onMounted(() => {
     multiSelectWithoutKeyboard(
         '#healthcare-professional-accepted-insurances',
-        healthcareProfessionalAcceptedInsurances,
+        submissionFormFields.healthcareProfessionalAcceptedInsurances,
         extractInsuranceOptions
     )
 
     multiSelectWithoutKeyboard(
         '#healthcare-professional-degrees',
-        healthcareProfessionalDegrees,
+        submissionFormFields.healthcareProfessionalDegrees,
         extractDegreeOptions
     )
 
     multiSelectWithoutKeyboard(
         '#healthcare-professional-specialties',
-        healthcareProfessionalSpecialties,
+        submissionFormFields.healthcareProfessionalSpecialties,
         extractSpecialtyOptions
     )
 
     multiSelectWithoutKeyboard(
         '#healthcare-professional-locales',
-        healthcareProfessionalLocales,
+        submissionFormFields.healthcareProfessionalLocales,
         extractLocaleOptions
     )
 })
 
-/* This handler function confirms that the user wants to return to the mod page, updates the modal state, updates the active
-screen and the visible url and router path so that the onBeforeRouteLeave function works as intended.
-If you don't update the route using push, the logic for the leave handler breaks.
-*/
-const handleConfirmationOfBack = () => {
+const handleNavigateToModerationScreen = () => {
```

Comment:
I tried to get this function to route the user to whatever their previous route was, regardless of it was `/moderation` or not. However, I found that other routes skipped the `onBeforeRouteLeave` logic entirely.

Reply:
If we used the `popState` and checked the `history` from the event, would we not be able to use a conditional to check if it is `moderation` and if it is to `router.push('/moderation')` and then `return` out otherwise to go back to the location pulled from the `popState` event?

Reply:
The issue isn't figuring out what the previous route is. We can get that from the `from` in `onBeforeRouteLeave`. What I observed is that none of the code will execute in this component if a users back to something other than `/moderation` like back to another submission they were viewing. It relaunches the auth, and none of the component logic is called.